core/frontend/web/site.js

```diff
@@ -171,6 +173,13 @@ module.exports = function setupSiteApp(options = {}) {
         }
     });
 
+    siteApp.use(function (req, res, next) {
+        if (req.member) {
+            DomainEvents.dispatch(MemberPageViewEvent.create({url: req.url, memberId: req.member.id}, new Date()));
```

Comment:
Passing a full member isn't a pattern I found in any events in `member-events`. What do you think about adding an extra `memberLastSeenAt` instead?
For example, `MemberPaidConversionEvent` uses `memberId` and `memberStatus` to pass data about a member: https://github.com/TryGhost/Members/blob/main/packages/member-events/lib/MemberPaidConversionEvent.js#L3-L4
I do like the idea of avoid database queries 👍

Reply:
That would work 👍🏼 I think it'll need a comment about why we're passing the last seen property, because it's not immediately obvious why it would be relevant for a MemberPageViewEvent